pkg/operator/ceph/cluster/cluster.go

```diff
@@ -52,6 +58,68 @@ func newCluster(c *cephv1beta1.Cluster, context *clusterd.Context) *cluster {
 		ownerRef: ClusterOwnerRef(c.Namespace, string(c.UID))}
 }
 
+func (c *cluster) detectCephMajorVersion() error {
+	// get the major ceph version by running "ceph --version" in the ceph image
+	job := &batch.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      detectVersionName,
+			Namespace: c.Namespace,
+		},
+		Spec: batch.JobSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"job": detectVersionName,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Command: []string{"ceph"},
+							Args:    []string{"--version"},
```

Comment:
What happens if this returns an error? There could be the (tiny) possiblity that some word in the error might match a valid Ceph version. Mostly I would expect `bash: ceph: command not found`, but I'm not sure what all might be possible with other unexpected errors. Would redirecting stderr to /dev/null here help protect Rook from that possibility (is that even possible with `Args`?)?

Reply:
If there is an unexpected error, i would hope the container would fail, in which case we don't look at the output. But I'll play with this a bit more to see if I can redirect stderr

Reply:
@travisn Status on this? As long as we make sure to check if the Pod is successfully exited before we get the pods logs we should be fine then.

Reply:
@BlaineEXE's separate PR for ceph to log to stderr will help avoid this issues of detecting the wrong output from this call.